src/include/gcc.h

```diff
@@ -248,12 +248,14 @@ WT_ATOMIC_FUNC(size, size_t, size_t *vp, size_t v)
     do {                                            \
         __asm__ volatile("dmb ishld" ::: "memory"); \
     } while (0)
-/*
- * This is necessary as we're moving to utilizing acquire and release semantics. The ARM load
- * barrier does provide us with acquire semantics but the store barrier does not. We upgrade it to a
- * full barrier for that reason.
- */
-#define WT_WRITE_BARRIER() WT_FULL_BARRIER();
+/* In order to achieve release semantics we need a both a load and store barrier. */
```

Comment:
we need both a load and a store barrier?

Reply:
Yes, load gives us LoadLoad LoadStore, store only give us StoreStore

Reply:
@quchenhao this is required for release semantics. We're renaming `WRITE_BARRIER` to `RELEASE_BARRIER` [in this PR](https://github.com/wiredtiger/wiredtiger/pull/10209/files#diff-08a92383c3904f531b067c488d6d6e34ddad0e3008313982b1b0712c0c3a7598R256).

The overall change we're making is the `WRITE_BARRIER` (StoreStore) is becoming a `RELEASE_BARRIER` (StoreStore, LoadStore), but since ARM doesn't have an instruction for release semantics we're choosing barriers that are stronger than release semantics.

Initially we weren't sure if separate `dmb ishld;  dmb ishst` was enough and defaulted to a `FULL_BARRIER` ([here](https://github.com/wiredtiger/wiredtiger/commit/b37a3618c50976c8673e909d12be1878ca02fa53)). Since then we've confirmed that the two `dmb` barriers are strong enough for release semantics, and the StoreLoad constraint from the `FULL_BARRIER` is just slowing things down.

Functionally the change before PM-3221 was
```
#define WT_WRITE_BARRIER()                          
    do {                                            
        __asm__ volatile("dmb ishst" ::: "memory"); 
    } while (0)
```
and after PM-3221 will be
```
#define WT_RELEASE_BARRIER()                          
    do {                                            
        __asm__ volatile(
            "dmb ishst" ::: "memory"; 
            "dmb ishld" ::: "memory"); 
    } while (0)
```

Reply:
We shouldn't use the terms "load barrier" and "store barrier" here. I think they make it more confusing. Instead say "dmb ld" and "dmb st". I think something like:

> We use dmb ld to provide the LoadStore ordering and dmb st to provide the StoreStore ordering. Together they prevent any memory access from before the barrier from happening reordering with any store after the barrier (we have confirmed this with Arm). dmb ld also provides LoadLoad order which isn't needed here. Using these two fences is faster than a full barrier because it avoids requiring the expensive StoreLoad barrier.

Reply:
It probably isn't worth mentioning in the comment, but the combination of dmb st + dmb ld provides the same ordering as power's `lwsync` instruction which we are using for both acquire and release on that platform.

Reply:
Removed the load ans store barrier. Just using dmb ishld and dmb ishst.